LunrCoreLmdb/SerializationExtensions.cs

```diff
@@ -0,0 +1,254 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Lunr;
+
+namespace LunrCoreLmdb
+{
+    public static class SerializationExtensions
+    {
+        #region Vector
+
+        public static ReadOnlySpan<byte> Serialize(this Vector vector)
+        {
+            var ms = new MemoryStream();
+            var bw =new BinaryWriter(ms);
+            var context = new SerializeContext(bw);
+
+            var values = vector.Save().ToList();
+            context.bw.Write(values.Count);
+            foreach (var value in values)
+            {
+                context.bw.Write(value);
+            }
+
+            ms.Position = 0;
+            return ms.GetBuffer();
+        }
+
+        public static Vector DeserializeFieldVector(this ReadOnlySpan<byte> buffer)
+        {
+            unsafe
+            {
+                fixed(byte* buf = &buffer.GetPinnableReference())
```

Comment:
I'm curious how much better this is than using `Span<T>`?

Reply:
I tried to answer this via: https://github.com/bleroy/lunr-core/pull/26/commits/4482cfd83507c070935393a0a8362de241e77c05

Note that the benchmark implementation is kind of wonky because I couldn't find an API that allowed me to construct a `Stream` from a `Span<byte>`, so I had to call the primitive methods directly.

It does look like Span is the clear winner, but I'm not confident in my implementation since it requires manually slicing the buffer.

``` ini

BenchmarkDotNet=v0.12.1, OS=Windows 10.0.18363.1256 (1909/November2018Update/19H2)
Intel Core i9-9820X CPU 3.30GHz, 1 CPU, 20 logical and 10 physical cores
.NET Core SDK=5.0.101
  [Host]     : .NET Core 3.1.10 (CoreCLR 4.700.20.51601, CoreFX 4.700.20.51901), X64 RyuJIT
  DefaultJob : .NET Core 3.1.10 (CoreCLR 4.700.20.51601, CoreFX 4.700.20.51901), X64 RyuJIT


```
|               Method |      Mean |    Error |   StdDev |
|--------------------- |----------:|---------:|---------:|
| GetPinnableReference | 583.48 ns | 2.811 ns | 2.629 ns |
|                 Span |  81.90 ns | 0.356 ns | 0.297 ns |

Reply:
That's pretty spectacular. Your implementation looks fine to me, but piece of mind should just be a matter of writing enough tests ;)